asyncgit/src/sync/utils.rs

```
@@ -144,6 +144,29 @@ pub fn stage_addremoved(repo_path: &str, path: &Path) -> Result<()> {
     Ok(())
 }
 
+/// get string from config
+pub fn get_config_string(
+    repo_path: &str,
+    key: &str,
+) -> Result<Option<String>> {
+    let repo = repo(repo_path)?;
+    let cfg = repo.config()?;
+
+    // this code doesnt match what the doc says regarding what
+    // gets returned when but it actually works
+    let entry_res = cfg.get_entry(key);
+
+    let entry = match entry_res {
+        Ok(ent) => ent,
+        Err(_) => return Ok(None),
+    };
+
+    if !entry.has_value() {
+        Ok(None)
+    } else {
+        Ok(entry.value().map(|s| s.to_string()))
+    }
```

I was thinking, could this function not just return a `ConfigOptionNotFound` error(a new error type that would need to be added) if the entry was not found, rather than none.  Not finding a config option in Git2rs is an error, I think.  With that in mind, an ultra succinct rust way to write this function could be:
```
    bytes2string(
        repo(repo_path)?
            .config()?
            .get_entry(key)
            .map_err(|_| error::ConfigOptionNotFound)?
            .value_bytes(),
    )
```
This also tests that the string is valid utf8 (using the function below), so we can throw a utf8 error, rather than a git2rs error.  I have not tested this but something like that should work.